### PR TITLE
[JENKINS-61178] When defining an extra container in a pod template with a custom working dir, it is wrongly used as the node working dir

### PR DESCRIPTION
**Pull request by Vlatombe**
[JENKINS-61178](https://issues.jenkins-ci.org/browse/JENKINS-61178)

### CODE REVIEW

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesSlave.java [hunk 1]
```diff
@@ -125,16 +129,13 @@ public KubernetesSlave(PodTemplate template, String nodeDescription, String clou
...
-                rs,
-                template.getNodeProperties());
-
+        super(name, null, computerLauncher);
+        setNodeDescription(nodeDescription);
+        setNumExecutors(1);
+        setMode(template.getNodeUsageMode() != null ? template.getNodeUsageMode() : Node.Mode.NORMAL);
+        setLabelString(labelStr);
+        setRetentionStrategy(rs);
+        setNodeProperties(template.getNodeProperties());
```
Comment (jglick) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesSlave.java]: Is this related, or just a deprecation warning?

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesLauncher.java [hunk 2]
```diff
@@ -110,9 +110,7 @@ public void launch(SlaveComputer computer, TaskListener listener) {
         try {
             KubernetesClient client = slave.getKubernetesCloud().connect();
             Pod pod = template.build(slave);
-            if (template.isShowRawYaml()) {
```
Comment (jglick) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesLauncher.java]: Related, or just another bug?

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesSlave.java (hunk 1 above)
Comment (jglick) [reply src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesSlave.java]: I guess you addressed the deprecation while dealing with the `remoteFs`?

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesSlave.java [hunk 3]
```diff
@@ -156,6 +157,34 @@ public String getPodName() {
...
+                if (optionalJnlp.isPresent()) {
+                    remoteFS = StringUtils.defaultIfBlank(optionalJnlp.get().getWorkingDir(), ContainerTemplate.DEFAULT_WORKING_DIR);
+                }
+            }
+        }
+        return Util.fixNull(remoteFS);
+    }
+
+    // Copied from Slave#getRootPath because this uses the underlying field
+    @CheckForNull
```
Comment (jglick) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesSlave.java]:
```suggestion
    @CheckForNull
    @Override
```

?

File: src/test/resources/org/csanchez/jenkins/plugins/kubernetes/pipeline/sidecarWorkingDir.groovy [hunk 4]
```diff
@@ -0,0 +1,12 @@
+podTemplate(containers: [
+    containerTemplate(name: 'busybox', image: 'busybox', ttyEnabled: true, command: '/bin/cat', workingDir: '/src')
```
Comment (jglick) [inline src/test/resources/org/csanchez/jenkins/plugins/kubernetes/pipeline/sidecarWorkingDir.groovy]: Should the test assert something about `/src`?

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesSlave.java (hunk 1 above)
Comment (Vlatombe) [reply src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesSlave.java]: Yes

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesLauncher.java (hunk 2 above)
Comment (Vlatombe) [reply src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesLauncher.java]: Not a bug, but with the impl change, having the pod at hand is mandatory, so this is combined with https://github.com/jenkinsci/kubernetes-plugin/pull/717/files#diff-7de256a4d98e2297b5323f04b76bd471R410

File: src/test/resources/org/csanchez/jenkins/plugins/kubernetes/pipeline/sidecarWorkingDir.groovy (hunk 4 above)
Comment (Vlatombe) [reply src/test/resources/org/csanchez/jenkins/plugins/kubernetes/pipeline/sidecarWorkingDir.groovy]: Right could be made slightly better.

Comment (Vlatombe) [reply src/test/resources/org/csanchez/jenkins/plugins/kubernetes/pipeline/sidecarWorkingDir.groovy]: Done
